client/src/pages/Calendar.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from "react";
+import Alert from "react-bootstrap/Alert";
+
+const Calendar = () => {
+	const [bookingsData, setBookingsData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		setIsLoading(true);
+		fetch("/api/bookings?filter=2weeks")
+			.then((response) => {
+				if (!response.ok) {
+					throw Error(
+						"There was an error getting the calendar data, Please try again"
+					);
+				}
+				return response.json();
+			})
+			.then((data) => {
+				setBookingsData(data);
+				setError(null);
+			})
+			.catch((err) => {
+				setError(err.message);
+			})
+			.finally(() => {
+				setIsLoading(false);
+			});
+	}, []);
+
+	if (isLoading) {
+		return (
+			<div className="d-flex justify-content-center">
+				<div className="spinner-grow">
+					<span className="visually-hidden">Loading...</span>
+				</div>
+			</div>
+		);
+	}
+	if (error != null) {
+		return <Alert variant="danger">{error}</Alert>;
+	}
+
+	const pickUpDates = [];
+
+	bookingsData.forEach((client) => {
+		let found = pickUpDates.find((el) => el.date == client.pick_up_date);
+		if (found) {
+			found.clients.push(client);
+		} else {
+			pickUpDates.push({ date: client.pick_up_date, clients: [client] });
+		}
+	});
```

Comment:
```suggestion
	});
	
	pickUpDates.sort((a, b) => new Date(a.date) - new Date(b.date));
```
Same as my other comment, we need to sort the array to show them on the UI in order

Reply:
Ok, already made.